inspirehep/modules/authors/static/js/authors/publications.js

```diff
@@ -0,0 +1,219 @@
+                date_parser(String(year_extent[1]+1))]).range([0, width - (margins * 2)]);
+
+            publicationsScale = d3.scale.linear()
+              .domain([0, d3.max(publicationsDataToPlot, function(d) {
+                return d.publications;
+              })])
+              .range([0, height]);
+
+            yearAxis = d3.svg.axis()
+              .scale(yearScale)
+              .orient("bottom")
+              .tickFormat(d3.time.format("%Y"));
+
+            publicationsAxis = d3.svg.axis()
+              .scale(publicationsScale)
+              .orient("left")
+              .ticks(4);
+
+            svg.append("svg:g")
+              .attr("class", "x axis")
+              .attr("transform", "translate(0," + (height) +")")
+              .call(yearAxis);
+
+            svg.append("svg:g")
+              .attr("class", "y axis")
+              .attr("transform", "translate(0, 0)")
+              .call(publicationsAxis);
+
+            svg.selectAll("publications")
+                .data(publicationsDataToPlot)
+              .enter().append("rect")
+                .style("fill", "rgb(13, 131, 183)")
+                .attr("x", function(d) { 
+                  return yearScale(date_parser(String(d.year))) - bar_width; })
+                .attr("width", bar_width)
+                .attr("y", height)
+                .attr("height", 0)
+                .transition()
+                  .duration(1000)
+                  .attr("y", function(d) {
+                    return (height) - publicationsScale(d.publications); })
+                  .attr("height", function(d) { 
+                    return publicationsScale(d.publications); })
+          }
+
+          if (publicationsDataToPlot.length != 0) {
+            drawBarChart();
+          }
+        });
+      }
+    };
+  });
+
+  app.directive('collaborators', function() {
+    return {
+      restrict: 'E',
+      templateUrl: '/static/js/authors/templates/collaborators.html',
+      scope: false,
+      controller: 'PublicationsController'
+    };
+  });
+
+  app.directive('publicationsList', function() {
+    return {
+      restrict: 'E',
+      templateUrl: '/static/js/authors/templates/publications.html',
+      scope: true,
+      controller: 'PublicationsController',
+      link: function(scope) {
+        scope.$watch('publications', function() {
+          scope.publicationsList = scope.publications.publications;
+
+          // Order of displayed publications.
+          scope.predicate = 'title';
+          scope.reverse = false;
+
+          scope.order = function(predicate) {
+            scope.reverse = (scope.predicate === predicate) ? !scope.reverse : false;
+            scope.predicate = predicate;
+          }
+        });
+      }
+    };
+  });
+
+    app.directive('statistics', function() {
+    return {
+      restrict: 'E',
+      templateUrl: '/static/js/authors/templates/statistics.html',
+      scope: true,
```

Comment:
Also related to the comment before of having a parent directive wrapping all the other ones. It could be a good idea to do it like in this [invenio library](https://github.com/inveniosoftware/invenio-search-js/blob/master/src/invenio-search-js/invenioSearch.module.js) used in search results page. 

There, all the directives that are inside of the parent just have `scope: False` and require the parent - see https://github.com/inveniosoftware/invenio-search-js/blob/master/src/invenio-search-js/invenioSearch.module.js#L558-L564 so that they all share the same controller and scope.